autoupgrade.php

```diff
@@ -149,9 +149,54 @@ public function uninstall()
         // Remove the 1-click upgrade working directory
         self::_removeDirectory(_PS_ADMIN_DIR_ . DIRECTORY_SEPARATOR . 'autoupgrade');
 
+        Configuration::deleteByName('PS_AUTOUP_IGNORE_REQS');
+        Configuration::deleteByName('PS_AUTOUP_IGNORE_PHP_UPGRADE');
+
         return parent::uninstall();
     }
 
+    /**
+     * Register the current module to a given hook and moves it at the first position.
+     *
+     * @param string $hookName
+     *
+     * @return bool
+     */
+    private function registerHookAndSetToTop($hookName)
+    {
+        return $this->registerHook($hookName) && $this->updatePosition((int) Hook::getIdByName($hookName), 0);
+    }
+
+    public function hookDashboardZoneOne($params)
+    {
+        // Display panel if PHP is not supported by the community
+        require_once __DIR__ . '/vendor/autoload.php';
+
+        $upgradeContainer = new \PrestaShop\Module\AutoUpgrade\UpgradeContainer(_PS_ROOT_DIR_, _PS_ADMIN_DIR_);
+        $upgrader = $upgradeContainer->getUpgrader();
+        $upgradeSelfCheck = new  \PrestaShop\Module\AutoUpgrade\UpgradeSelfCheck(
+            $upgrader,
+            _PS_ROOT_DIR_,
+            _PS_ADMIN_DIR_,
+            __DIR__
+        );
+
+        $upgradeNotice = $upgradeSelfCheck->isPhpUpgradeRequired();
+        if (false === $upgradeNotice) {
+            return false;
```

Comment:
return empty string instead?

Reply:
I hesitated on this one as well, but it would seem more logical indeed